File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 1; i<=200000; i++) {
            map.put(i, 0);
        }
        
        for (int i = 0; i<nums.length; i++) {
            if (map.get(nums[i]) == 0) {
                map.put(nums[i], 1);
                answer++;
            }
            
            if (answer == nums.length/2) {
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    map = [0] * 200001
    
    for i in range(len(nums)):
        if (map[nums[i]] == 0):
            map[nums[i]] = 1
            answer+=1
            
        if len(nums) // 2 == answer:
            break
    return answer